sys/include/crypto/aes.h

```diff
@@ -37,6 +37,8 @@ typedef uint32_t u32;
 typedef uint16_t u16;
 typedef uint8_t u8;
 
+/* This controls AES table calculation on the fly */
+#define AES_CALCULATE_TABLES
```

Comment:
define wrapped.

> Here is what I did to verify this is correct.

It passes unittests/tests-crypto/tests-crypto-aes.c which means it is correct.

Reply:
D'oh! I just realized what I wrote. I was thinking on defines that take on values, but this is another thing.

According to @cladmi the correct way of handling this is by using a PSEUDOMODULE. I just did it, you can look for it in my branch https://github.com/jcarrano/RIOT/tree/aes-t-table

> It passes unittests/tests-crypto/tests-crypto-aes.c which means it is correct.

Sure, but that tests compression and decompression with the same code. I wanted to check that both codes were equivalent (and on the process set a precedent so that in the future when people make big but mechanical changes, I can direct them to this PR to show them how they can help me review it).

Reply:
Pseudomodules are ok, except I don't think that precalculated tables nor unrolled cycles is a good default choice. Both are extremely costly in terms of flash memory (I don't even think that anyone ever used FULL_UNROLL option).

> Sure, but that tests compression and decompression with the same code

No. It tests actual code against so called test vector — which include both input, encryption key and output. Any AES code under test must provide specified output being given specified key and input. 

There are two different test vectors defined, so false positive is near to impossible. They are not [NIST recommended vectors](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38a.pdf), but that should not matter if vectors were calculated properly.

Reply:
> except I don't think that precalculated tables nor unrolled cycles is a good default choice

Me neither. That's why those features won't be selected unlesss one explicitly selects the pseudomodule. If you just type `USEMODULE += crypto` then tables will be on-the-fly (I changed the ifdef to ifndef in my branch) and loops rolled.